scripts/cat/skills.py

```diff
@@ -361,21 +361,19 @@ def __repr__(self) -> str:
         return f"<CatSkills: Primary: |{self.primary}|, Secondary: |{self.secondary}|, Hidden: |{self.hidden}|>"
 
     @staticmethod
-    def generate_new_catskills(rank, moons, hidden_skill: HiddenSkillEnum = None):
+    def generate_new_catskills(
+        rank: CatRank, age: CatAge, hidden_skill: HiddenSkillEnum = None
+    ):
         """Generates a new skill"""
         new_skill = CatSkills()
 
         new_skill.hidden = hidden_skill
 
-        if rank == CatRank.NEWBORN or moons <= 0:
+        if rank == CatRank.NEWBORN or CatAge.NEWBORN:
             pass
-        elif rank == CatRank.KITTEN or moons < 6:
+        elif rank == CatRank.KITTEN or CatAge.KITTEN:
             new_skill.primary = Skill.get_random_skill(points=0, interest_only=True)
-        elif rank in [
-            CatRank.APPRENTICE,
-            CatRank.MEDICINE_APPRENTICE,
-            CatRank.MEDIATOR_APPRENTICE,
-        ]:
+        elif rank.is_any_apprentice_rank() or CatAge.ADOLESCENT:
```

Comment:
```suggestion
        elif rank.is_any_apprentice_rank() or age == CatAge.ADOLESCENT:
```

Reply:
I do wonder whether this messes with early graduates, but this init should only be called with fresh generated cats right?